keras/layers/wrappers.py

```diff
@@ -201,22 +218,41 @@ def step(x, _):
             self._input_map[input_uid] = inputs
             # (num_samples * timesteps, ...)
             y = self.layer.call(inputs, **kwargs)
-            if hasattr(y, '_uses_learning_phase'):
-                uses_learning_phase = y._uses_learning_phase
             # Shape: (num_samples, timesteps, ...)
             output_shape = self.compute_output_shape(input_shape)
-            y = K.reshape(y, (-1, input_length) + output_shape[2:])
+
+            # handle multiple outputs
+            if not isinstance(output_shape, list):
+                output_shape = [output_shape]
+                y = [y]
+
+            if LPWrapper.uses_learning_phase is False:
+                LPWrapper.uses_learning_phase = []
+            for idx, yi in enumerate(y):
+                if hasattr(yi, '_uses_learning_phase'):
+                    LPWrapper.uses_learning_phase.append(yi._uses_learning_phase)
+                else:
+                    LPWrapper.uses_learning_phase.append(False)
+
+            y = [K.reshape(yi, (-1, input_length) + output_shape[idx][2:])
+                 for idx, yi in enumerate(y)]
 
         # Apply activity regularizer if any:
         if (hasattr(self.layer, 'activity_regularizer') and
            self.layer.activity_regularizer is not None):
-            regularization_loss = self.layer.activity_regularizer(y)
-            self.add_loss(regularization_loss, inputs)
-
-        if uses_learning_phase:
-            y._uses_learning_phase = True
+            regularization_losses = [self.layer.activity_regularizer(yi) for yi in y]
+            self.add_loss(regularization_losses, inputs)
+
+        for idx, yi in enumerate(y):
+            if LPWrapper.uses_learning_phase[idx]:
+                yi._uses_learning_phase = True
+        if len(y) == 1:
+            y = y[0]
         return y
 
+    def compute_mask(self, inputs, mask=None):
+        return self.layer.compute_mask(inputs, mask)
```

Comment:
Indeed, I noticed the bug a moment ago. :(

Multi-output layers don't work unless a custom mask is defined, so I thought the same would apply to a TD layer. It seems I was wrong. Could you tell me why that's not the case here?

I have a fix that is similar to what call does, as in the following:

```
def compute_mask(self, inputs, mask=None):
    input_shape = K.int_shape(inputs)
    inputs = K.reshape(inputs, (-1,) + input_shape[2:])
    return self.layer.compute_mask(inputs, mask)
```

but I'll have to check it out more exhaustively tomorrow to find out whether it's required for my use case.

For the time being, I'll just remove the method altogether since the PR seems to pass all the tests without it. What do you think?